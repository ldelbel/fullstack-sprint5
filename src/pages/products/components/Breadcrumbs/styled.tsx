import styled from "styled-components";

export const StyledBreadcrumbs = styled.section`
  margin-bottom: 16px;

  nav {
    max-width: 1200px;
    margin: 0 auto;
  }

  .breadcrumbs__list {
    display: flex;
  }

  @media (max-width: 1200px) {
    .breadcrumbs__list {
      justify-content: center;
    }
  }
`;
