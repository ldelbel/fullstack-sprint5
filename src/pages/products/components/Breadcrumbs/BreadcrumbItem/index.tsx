import React from 'react';
import { StyledBreadcrumbItem } from "./styled";

interface IBreadcrumbItem {
  link: string;
  name: string;
}

export function BreadcrumbItem({ link, name }: IBreadcrumbItem) {
  return (
    <StyledBreadcrumbItem>
      {link ? (
        <>
          <a className="breadcrumbs__link" href={link}>
            {name}
          </a>
          <span className="breadcrumbs__item  breadcrumbs__separator">/</span>
        </>
      ) : (
        <span className="breadcrumbs__link">{name}</span>
      )}
    </StyledBreadcrumbItem>
  );
}