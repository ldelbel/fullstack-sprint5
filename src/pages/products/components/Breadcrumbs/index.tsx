import { useContext } from "react";
import CategoriesContext from "../../../../contexts/CategoriesContext";
import { ICategoriesContext } from "../../../../types/interfaces";
import { BreadcrumbItem } from "./BreadcrumbItem";
import { StyledBreadcrumbs } from "./styled";

function Breadcrumbs() {
  const { categories } = useContext<ICategoriesContext>(CategoriesContext);

  return (
    <StyledBreadcrumbs>
      <nav>
        <ol className="breadcrumbs__list">
          {categories.current &&
            categories.current.map((c) => (
              <BreadcrumbItem key={c.id} link={c.link} name={c.name} />
            ))}
        </ol>
      </nav>
    </StyledBreadcrumbs>
  );
}

export default Breadcrumbs;
