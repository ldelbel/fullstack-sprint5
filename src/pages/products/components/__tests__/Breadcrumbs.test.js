import React from "react";
import Breadcrumbs from "../Breadcrumbs";
import ShallowRenderer from "react-test-renderer/shallow";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react", () => {
  const react = jest.requireActual("react");
  return {
    ...react,
    useContext: jest.fn(),
  };
});

const mockContext = {
  categories: {
    current: [
      {
        id: 1,
        link: "#home",
        name: "Home",
      },
      {
        id: 2,
        link: "#home",
        name: "Infantil",
      },
    ],
  },
};

describe("Breadcrumbs component", () => {

  it("should render breadcrumbs", () => {
    React.useContext.mockImplementation(() => {
      return mockContext;
    });

    const breadcrumbs = Enzyme.shallow(<Breadcrumbs />);
    expect(breadcrumbs.find("ol").hasClass("breadcrumbs__list")).toBeTruthy();
  });

  it("should match snapshot", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Breadcrumbs />);
    const breadcrumbs = renderer.getRenderOutput();
    expect(breadcrumbs).toMatchSnapshot();
  });
  
});
