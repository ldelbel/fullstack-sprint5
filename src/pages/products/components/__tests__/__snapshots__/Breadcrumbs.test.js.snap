// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should match snapshot 1`] = `
<section
  class="sc-gtsrHT oqtMB"
>
  <nav>
    <ol
      class="breadcrumbs__list"
    >
      <div
        class="sc-bdnxRM dhyORQ"
      >
        <a
          class="breadcrumbs__link"
          href="#home"
        >
          Home
        </a>
        <span
          class="breadcrumbs__item  breadcrumbs__separator"
        >
          /
        </span>
      </div>
      <div
        class="sc-bdnxRM dhyORQ"
      >
        <a
          class="breadcrumbs__link"
          href="#home"
        >
          Infantil
        </a>
        <span
          class="breadcrumbs__item  breadcrumbs__separator"
        >
          /
        </span>
      </div>
    </ol>
  </nav>
</section>
`;
